Add profile for ARM compatibility (#1425)
nf-core/rnaseq: docs/usage.md

## Alignment options

:::note

> Not currently available using ARM architecture ('-profile arm')

Reviewer: ```suggestion
The `--aligner hisat2` option is not currently supported using ARM architecture ('-profile arm')
```

## Contamination screening options

:::note

> Not currently available using ARM architecture ('-profile arm')

Reviewer: Maybe say specifically which parameters aren't supported

You can also generate such `YAML`/`JSON` files via [nf-core/launch](https://nf-co.re/launch).

### Running on Linux ARM architectures

The pipeline can be executed in an ARM compatible mode by specifying the ARM profile, for example:

```bash
nextflow run \
    nf-core/rnaseq \
    --input <SAMPLESHEET> \
    --outdir <OUTDIR> \
    --gtf <GTF> \
    --fasta <GENOME FASTA> \
    -profile docker,arm
```

> This will use ARM-compatible containers, and apply a small number of overrides to Conda definitions to support ARM operation.

Reviewer: ```suggestion
This will use ARM-compatible containers, and apply a small number of overrides to Conda definitions to support ARM operations.
```

:::warning
Please note that the ARM profile is experimental. It is expected to function correctly in all cases unless explicitly indicated otherwise—currently, exceptions include the use of the hisat2 aligner and contaminant screening via kraken2. However, because testing is presently conducted manually, we cannot guarantee its reliability.
:::

> 

Reviewer: Move this to the top of this section

  - A generic configuration profile to enable [Wave](https://seqera.io/wave/) containers. Use together with one of the above (requires Nextflow ` 24.03.0-edge` or later).
- `conda`
  - A generic configuration profile to be used with [Conda](https://conda.io/docs/). Please only use Conda as a last resort i.e. when it's not possible to run the pipeline with Docker, Singularity, Podman, Shifter, Charliecloud, or Apptainer.

>   - A configuration profile that will set `docker.runOptions` appropriately for ARM architectures, and apply overrides supplying ARM-compatible containers and Conda environments.

Reviewer: Add an explicit section for `-profile arm` and clarify that it is an experimental feature.

  - A generic configuration profile to enable [Wave](https://seqera.io/wave/) containers. Use together with one of the above (requires Nextflow ` 24.03.0-edge` or later).
- `conda`
  - A generic configuration profile to be used with [Conda](https://conda.io/docs/). Please only use Conda as a last resort i.e. when it's not possible to run the pipeline with Docker, Singularity, Podman, Shifter, Charliecloud, or Apptainer.
- `arm`
  - A configuration profile that will set `docker.runOptions` appropriately for ARM architectures, and apply overrides supplying ARM-compatible containers and Conda environments.

> :::warning
> Please note that the ARM profile is experimental. It is expected to function correctly in all cases unless explicitly indicated otherwise—currently, exceptions include the use of the hisat2 aligner and contaminant screening via kraken2. However, because testing is presently conducted manually, we cannot guarantee its reliability.
> :::
> 

Reviewer: ```suggestion
```

Reviewer: Maybe we just link to the new section we created in the `arm` profile above? Means we don't need to repeat the text.
